File: lib/model/empresa_model.dart
class EmpresaModel {
  late int id;
  late DateTime dataCriacao;
  late String cnpj;
  late String endereco;
  late String razaoSocial;
  late String nome;
  late String contato;
  late String fotoEmpresa;

  EmpresaModel({required this.id,required this.dataCriacao,required this.cnpj,required this.endereco,required this.razaoSocial,
  required this.nome, required this.fotoEmpresa, required this.contato});

  EmpresaModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    dataCriacao = DateTime.parse(json['dataCriacao']);
    cnpj = json['cnpj'];
    endereco = json['endereco'];
    razaoSocial = json['razaoSocial'];
    nome = json['nome'];
    contato = json['contato'];
    fotoEmpresa = json['fotoEmpresa'];
  }
}

File: android/app/src/main/kotlin/com/exercicio/exercicio/EmpresaModel.kt
package com.exercicio.exercicio

class EmpresaModel {
    constructor(id: Int ,cnpj: String, endereco: String?, razaoSocial: String?, nome: String?, contato: String?, fotoEmpresa: String?, dataCriacao: String?) {
        this.id = id
        this.cnpj = cnpj
        this.endereco = endereco
        this.razaoSocial = razaoSocial
        this.nome = nome
        this.contato = contato
        this.fotoEmpresa = fotoEmpresa
        this.dataCriacao = dataCriacao
    }

    var id = 0
    var dataCriacao: String? = null
    var cnpj: String? = null
    var endereco: String? = null
    var razaoSocial: String? = null
    var nome: String? = null
    var contato: String? = null
    var fotoEmpresa: String? = null
}